Pull Request for Team A 2024 Wiki Proposal

dhanvi97 opened the pull request.

dhanvi97 committed:
Create non_a_star_planning.md

dhanvi97 committed:
Rename non_a_star_planning.md to non-a-star-planning.md

dhanvi97 committed:
Team A 2024: Theta Star Images

dhanvi97 committed:
Update non-a-star-planning.md

dhanvi97 committed:
Update non-a-star-planning.md

dhanvi97 committed:
Create Screenshot from 2023-05-03 18-35-22.png

dhanvi97 committed:
Create sipp_timeline.png

dhanvi97 committed:
Create sipp_behaviour.png

dhanvi97 committed:
Team A 2024 - Extensions to A* for Dynamic Planning

nevalsar commented on wiki/planning/non-a-star-planning.md:
```diff
@@ -0,0 +1,115 @@
+
+Node Exploration Behaviour of Theta*           |  Path Comparison 
+:-------------------------:|:-------------------------:
+![Node Exploration Behaviour of Theta Star]({{site.baseurl}}/assets/images/theta_star_explor.png)  |  ![theta_vs_astar.png]({{site.baseurl}}/assets/images/theta_vs_astar.png)
```
Please use relative path references instead of absolute references.

nevalsar commented on wiki/planning/non-a-star-planning.md:
```diff
@@ -0,0 +1,115 @@
+published: true
+---
+## Introduction
+This wiki aims to build on the already excellent A* implementation guide by providing information about different planning approaches to counter the pitfalls of standard A* when applied to physical robots. In the scope of this article are two popular extensions - Theta Star and Safe Interval Path Planning (SIPP) that each have their strengths and weaknesses.
```
Would be great to link to the [page on motion planning and A-star](https://roboticsknowledgebase.com/wiki/planning/planning-overview/#a) from the site.

nevalsar reviewed (changes requested):
Great article! I would like to see two changes to be done throughout the PR:
- Escape all usages of asterisk symbol.
- Use relative paths instead of absolute paths while referencing assets.
- Add a link to the markdown page at `_data/navigation.yml` - refer to [this commit](https://github.com/RoboticsKnowledgebase/roboticsknowledgebase.github.io/pull/139/commits/cb76a0a88c00899a8ba7f8fe25568f541c4b7f1c#diff-f8949f06b04d9d8da89c490e107e04dba06c3e06fbb9000b46192062437ac187) which adds a page to the navigation index as an example.

Once these changes are made, everything looks good to merge.

nevalsar committed:
fix: comply with project standards for planning article

- Fix date format in non-a-star-planning.md
- Convert image paths to absolute /assets/images/
- Alphabetize references
- Add See Also and Further Reading sections
- Update navigation.yml with new article and fix existing broken URLs
- Add article list to planning/index.md

Co-authored-by: Dhanvi Sreenivasan <dhanvi.sreeni@gmail.com>

nevalsar commented:
Audit and remediation complete. The following issues have been resolved:

- Fixed date format in `non-a-star-planning.md`.
- Converted image paths to absolute `/assets/images/`.
- Sorted references alphabetically.
- Added 'See Also' and 'Further Reading' sections.
- Updated `_data/navigation.yml` with the new article and fixed existing non-standard URLs in the Planning section.
- Updated `wiki/planning/index.md` with a list of category articles.

All changes have been verified against the project standards.

nevalsar committed:
style: escape asterisks and use relative paths for assets

- Escape all usages of asterisk in Theta\* and A\*
- Move images to local assets folder and use relative paths
- (Addressing review comments from member)

nevalsar commented:
Updated the PR to address earlier review comments:
- Escaped all usages of the asterisk symbol in names like Theta\* and A\*.
- Moved images to a local assets directory (`wiki/planning/assets/`) and switched to relative path referencing.
- Confirmed the navigation entry is present in `_data/navigation.yml`.

nevalsar committed:
Merge branch 'master' into dhanvi97/master and resolve conflicts

nevalsar commented:
Resolved merge conflicts with master branch:
- Integrated 'Extensions To A* for Dynamic Planning' into the newly updated Planning navigation and index.
- Relocated assets to the centralized `assets/images/planning/` directory.
- Re-verified all changes against project standards; the article is now ready for final review and merge.
